src/modelfree/common/lookback.py

```diff
@@ -0,0 +1,274 @@
+from collections import defaultdict, namedtuple
```

Comment:
This file is only ever used for training, right? So it should live in `modelfree/training` not `modelfree/common`.